### PR TITLE
Rethrow SchemaManagerExceptions and InvalidOperationExceptions in SqlSchemaManager::ApplySchema

### PR DESCRIPTION
**Pull request by adrianwyatt**
## Description
Rethrow SchemaManagerExceptions and InvalidOperationExceptions in SqlSchemaManager::ApplySchema to prevent false successes in integrated/managed environments.

## Related issues
DevOps bug #81336 - "Service Fabric Schema Job reports success even when it fails"

## Testing
- Added unit tests
- Validated schema upgrade failure detected in managed/integrated environments.

## Semver Change ([docs](https://github.com/microsoft/healthcare-shared-components/blob/master/docs/Versioning.md))
Skip

### CODE REVIEW

File: test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs [hunk 1]
```diff
@@ -101,5 +103,31 @@ public async Task ApplySchema_UsingDiffScript_Succeeds()
             await _sqlSchemaManager.ApplySchema("connectionString", new Uri("https://localhost/"), new MutuallyExclusiveType { Latest = false, Version = 2, Next = false });
             await _schemaManagerDataStore.Received().ExecuteScriptAndCompleteSchemaVersionAsync(Arg.Is("script"), Arg.Is(2), Arg.Any<CancellationToken>());
         }
+
+        [Fact]
+        public async Task ApplySchema_OnDependencyThrowSchemaManagerException_ThrowsSchemaManagerException()
+        {
+            // Set a zero retry sleep duration to expedite fail-case unit test.
+            FieldInfo retrySleepDurationField = typeof(SqlSchemaManager).GetField("_retrySleepDuration", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (Ivanidzo4ka) [inline test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs:111]: >_retrySleepDuration [](http://example.com/codeflow?start=83&length=19)

Wouldn't it be better to make this part of configuration instead doing reflection magic?
`SqlServerSchemaOptions` sounds like good place to hold it.

Comment (adrianwyatt) [reply test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs:111]: I considered that, but with this pattern I am avoiding having a requirement that is specific to unit tests leak into the public API.

Comment (adrianwyatt) [reply test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs:111]: @Ivanidzo4ka, Please re-open if you feel strongly and I will update it to use SqlServerSchemaOptions.

File: test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs [hunk 2]
```diff
@@ -101,5 +103,31 @@ public async Task ApplySchema_UsingDiffScript_Succeeds()
...
+
+        [Fact]
+        public async Task ApplySchema_OnDependencyThrowInvalidOperationException_ThrowsInvalidOperationException()
+        {
+            _schemaManagerDataStore.GetCurrentSchemaVersionAsync(default).ReturnsForAnyArgs(Task.FromResult(1));
+            _client.GetCurrentVersionInformationAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(Task.FromException<List<CurrentVersion>>(new InvalidOperationException()));
+            _client.GetAvailabilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new List<AvailableVersion> { new AvailableVersion(1, "_script/1.sql", "_script/1.diff.sql"), new AvailableVersion(2, "_script/2.sql", "_script/2.diff.sql") });
+            _client.GetCompatibilityAsync(Arg.Any<CancellationToken>()).ReturnsForAnyArgs(new CompatibleVersion(1, 2));
+            _client.GetDiffScriptAsync(Arg.Is<Uri>(new Uri("_script/2.diff.sql", UriKind.Relative)), Arg.Any<CancellationToken>()).Returns("script");
+            await Assert.ThrowsAsync<InvalidOperationException>(async () => await _sqlSchemaManager.ApplySchema("connectionString", new Uri("https://localhost/"), new MutuallyExclusiveType { Latest = false, Version = 2, Next = false }));
```
Comment (YazanMSFT) [inline test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs:130]: Just FYI, I don't think the inner async/await are necessary - I think Assert.ThrowsAsync internally will do that

**Review by YazanMSFT — APPROVED** [review]

File: test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs (hunk 2 above)
Comment (adrianwyatt) [reply test/SchemaManager.Core.UnitTests/SqlSchemaManagerTests.cs]: Updated.

**Pull request merged**